lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1660,6 +1660,78 @@ struct HandshakeAnalysisPass
 };
 } // namespace
 
+// This pass iteratively checks the use count of each value, and reformats the
+// MLIR such that every value has exact one use. If a value has multiple uses, a
+// fork is inserted. If a value has no use, a sink is inserted.
+void insertOpForExactOneUse(Value value, handshake::FuncOp funcOp,
+                            OpBuilder &rewriter) {
+
+  if (value.use_empty())
+    rewriter.create<SinkOp>(funcOp.getLoc(), value);
```

Comment:
I'd recommend a "return" early exit after this so you don't need an "else" and the associated indentation.